pkg/apis/pipeline/v1beta1/task_validation.go

```diff
@@ -329,7 +330,7 @@ func (p ParamSpec) ValidateObjectType() *apis.FieldError {
 }
 
 // ValidateParameterVariables validates all variables within a slice of ParamSpecs against a slice of Steps
-func ValidateParameterVariables(ctx context.Context, steps []Step, params []ParamSpec) *apis.FieldError {
+func ValidateParameterVariables(ctx context.Context, steps []Step, params []ParamSpec, skipVariableValidation bool) *apis.FieldError {
```

Comment:
With the flag bool variable, `ctx` is not needed anymore for its downstream callees right?

Reply:
we need it downstream for some other calls. eg. validateVariable